resources/custom-resources/keycloak-setup/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/KeycloakSetup.java

```diff
@@ -270,7 +285,72 @@ protected int postRealm(String keycloakHost, String bearerToken, Map<String, Obj
         }
     }
 
-    protected int mapAdminUserServiceRoles(String keycloakHost, String bearerToken, String realmName, String username,
+    protected String createAdminGroup(String keycloakHost, String bearerToken, String realm, String groupName) {
+        try {
+            URI endpoint = new URI(keycloakHost + "/admin/realms/" + realm + "/groups");
+            HttpRequest request = HttpRequest.newBuilder()
+                    .version(HttpClient.Version.HTTP_1_1)
+                    .uri(endpoint)
+                    .setHeader("Authorization", "Bearer " + bearerToken)
+                    .setHeader("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(Utils.toJson(Map.of("name", groupName))))
+                    .build();
+            LOGGER.info("Invoking Keycloak group create endpoint {}", request.uri());
+            HttpResponse<String> response = httpClient.send(request,
+                    HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8));
+            if (response.statusCode() == HttpURLConnection.HTTP_CREATED) {
+                // let's return the group id we just created
+                Map<String, Object> group = findGroupByName(keycloakHost, bearerToken, realm, groupName);
+                String id = (String) group.get("id");
+                if (id != null) {
+                    return id;
+                }
+                throw new RuntimeException("Unexpected error: created group did not have id: " + Utils.toJson(group));
+            } else {
+                LOGGER.error("Expected HTTP_CREATED ({}) from group create, but got {}",
+                        HttpURLConnection.HTTP_CREATED, response.statusCode());
+                throw new RuntimeException("Unexpected error while creating group: " + groupName);
+            }
+        } catch (URISyntaxException | IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected Map<String, Object> findGroupByName(String keycloakHost, String bearerToken,
```

Comment:
Might be able to do GET /{realm}/groups?search={groupName} here instead of pulling back all the groups and iterating over them

Reply:
Nice callout, I'll do that instead.